### PR TITLE
sinon: add overrides parameter to createStubInstance

### PR DESCRIPTION
**Pull request by aleung**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://sinonjs.org/releases/v7.2.4/stubs/
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by nicojs** [comment]
Looks good. For ts 3.1, it would also be possible to type the `overrides` without using `any` using something like this:

```ts

        createStubInstance<TType>(
            constructor: StubbableType<TType>,
            overrides?: { [K in keyof TType]?: SinonStubbedMember<TType[K]> |  TType[K] extends (...args: any[]) => infer R ? R : TType[K] }
        ): SinonStubbedInstance<TType>;
```

**Comment by aleung** [comment]
@nicojs Thanks, will update.

If I only change the `index.d.ts` in `ts3.1` folder, does it apply to 3.2, 3.3 ... ?

**Comment by 43081j** [comment]
Yes it does

**Comment by nicojs** [comment]
👍 lgtm

**Review by nicojs — APPROVED** [review]
LGTM 👍

File: types/sinon/ts3.1/sinon-tests.ts [hunk 1]
```diff
@@ -87,6 +87,10 @@ function testSandbox() {
     const privateFooFoo: sinon.SinonStub = privateFooStubbedInstance.foo;
     const clsBar: number = stubInstance.bar;
     const privateFooBar: number = privateFooStubbedInstance.bar;
+    sb.createStubInstance(cls, {
+        foo: (arg1: string, arg2: number) => 2,
```
Comment (felixfbecker) [inline types/sinon/ts3.1/sinon-tests.ts]: This test doesn't doesn't look correct. According to the docs, the value should either be a stub object or a return value to be used, i.e. here `SinonStub | number`, but it is a function

**Pull request merged**